src/sidechaindb.cpp

```diff
@@ -106,148 +134,172 @@ std::vector<SidechainDeposit> SidechainDB::GetDeposits(uint8_t nSidechain) const
+        // Print sidechain name
+        str += "Sidechain: " + s.GetSidechainName() + "\n";
+        // Print sidechain WT^ workscore(s)
+        std::vector<SidechainWTJoinState> vState = GetState(s.nSidechain);
+        for (const SidechainWTJoinState& state : vState) {
+            str += "WT^: " + state.wtxid.ToString() + "\n";
+            str += "workscore: " + std::to_string(state.nWorkScore) + "\n";
+        }
+        str += "\n";
+    }
+    return str;
 }
 
 bool SidechainDB::Update(int nHeight, const uint256& hashBlock, const std::vector<CTxOut>& vout, std::string& strError)
 {
     if (hashBlock.IsNull())
         return false;
+    if (!vout.size())
+        return false;
+
+    // TODO skip if nHeight < drivechains activation block height
 
     // If a sidechain's tau period ended, reset WT^ verification status
     for (const Sidechain& s : ValidSidechains) {
         if (nHeight > 0 && (nHeight % s.GetTau()) == 0)
             SCDB[s.nSidechain].ClearMembers();
     }
+    // TODO clear out cached WT^(s) that belong to the Sidechain
+    // that was just reset.
 
     /*
-     * Only one state script of the current version is valid.
-     * State scripts with invalid version numbers will be ignored.
-     * If there are multiple state scripts with valid version numbers
-     * the entire coinbase will be ignored by SCDB and a default
-     * ignore vote will be cast. If there isn't a state update in
-     * the transaction outputs, a default ignore vote will be cast.
+     * Now we will look for data that is relevent to SCDB
+     * in this block's coinbase.
+     *
+     * Scan for h* linking data and add it to the BMMLD
+     * ratchet system.
+     *
+     * Scan for new WT^(s) and start tracking them.
+     *
+     * Scan for updated SCDB MT hash, and perform MT hash
+     * based SCDB update.
+     *
+     * Update hashBlockLastSeen to reflect that we have
+     * scanned this latest block.
      */
 
-    // Scan for state script
-    std::vector<CScript> vStateScript;
-    for (const CTxOut& out : vout) {
-        const CScript& scriptPubKey = out.scriptPubKey;
-
-        // Minimum size
-        if (scriptPubKey.size() < 3)
-            continue;
-        // State script begins with OP_RETURN
-        if (!scriptPubKey.IsUnspendable())
-            continue;
-        // Check state script version
-        if (scriptPubKey[1] != SCOP_VERSION || scriptPubKey[2] != SCOP_VERSION_DELIM)
-            continue;
-
-        vStateScript.push_back(scriptPubKey);
-    }
-
-    if (vStateScript.size() == 1 && ApplyStateScript(vStateScript[0], true)) {
-        ApplyStateScript(vStateScript[0]);
-    } else {
-        strError = "SidechainDB::Update: failed to apply state script\n";
-        ApplyDefaultUpdate();
-    }
-
-    // Scan for h*(s) in coinbase outputs
+    // TODO use CScript::IsBribeHashCommit() function once
+    // format is finalized.
+    // Scan for h*(s)
     for (const CTxOut& out : vout) {
```

Comment:
This is confusing too me -- why is `h*` relevant here? Isn't this adding outputs to `WT^` to pay out people wishing to withdraw from the sidechain?